room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -101,6 +120,21 @@ public synchronized SupportSQLiteDatabase getReadableDatabase() {
         if (!mVerified) {
             verifyDatabaseFile();
             mVerified = true;
+            if(mDatabaseConfiguration.prepackagedCallbacks != null) {
```

Comment:
can we extract this into another method to avoid code duplication?

Reply:
I wanted to extract this logic into a separate method and call it in `SQLiteCopyOpenHelper.getWritableDatabase()` and `SQLiteCopyOpenHelper.getReadableDatabase()` but when I temporarily open the db using `FrameworkSqliteOpenHelper` then I need to call `getReadableDatabase()` or `getWritableDatabase()` and I wasn't sure if I can just use one of those for both calls.
For example - can I open the db using framework helper, call `getWritableDatabase()` and pass it to the `onPrepackagedCallback` inside SQLiteCopyOpenHelper.getReadableDatabase()? Or I have to call `getReadableDatabase()` in this case?

Reply:
You can still extract the logic into a function that receives a boolean parameter `writable` to know which of the two methods to call.